Makefile

```diff
@@ -177,7 +177,7 @@ lint-gosec:
 .PHONY: lint-staticcheck
 lint-staticcheck:
 	@cat /dev/null > $(LINT_LOG)
-	time staticcheck ./... \
+	time staticcheck -checks=-SA1019 ./... \
```

Comment:
> SA1019 – Using a deprecated function, variable, constant or field

The preferred way is to only exclude specific occurrences in the code, via linter comment, not across the whole project. What's tripping this check? Can we fix that instead of silencing?

Reply:
```
time staticcheck ./... \
	| grep -v \
		-e model/model.pb.go \
		-e thrift-gen/ \
	>> .lint.log || true
Detected staticcheck failures:
cmd/collector/app/zipkin/http_handler.go:30:2: Package github.com/golang/protobuf/proto is deprecated: Use the "google.golang.org/protobuf/proto" package instead.  (SA1019)
cmd/collector/app/zipkin/http_handler_test.go:30:2: Package github.com/golang/protobuf/proto is deprecated: Use the "google.golang.org/protobuf/proto" package instead.  (SA1019)
model/prototest/model_test.pb.go:8:2: Package github.com/golang/protobuf/proto is deprecated: Use the "google.golang.org/protobuf/proto" package instead.  (SA1019)
model/prototest/model_test.pb.go:41:9: proto.EnumName is deprecated: Do not use.  (SA1019)
model/prototest/model_test.pb.go:65:9: xxx_messageInfo_SpanRef.Unmarshal is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
model/prototest/model_test.pb.go:68:9: xxx_messageInfo_SpanRef.Marshal is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
model/prototest/model_test.pb.go:71:2: xxx_messageInfo_SpanRef.Merge is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
model/prototest/model_test.pb.go:74:9: xxx_messageInfo_SpanRef.Size is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
model/prototest/model_test.pb.go:77:2: xxx_messageInfo_SpanRef.DiscardUnknown is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
model/prototest/model_test.pb.go:80:29: proto.InternalMessageInfo is deprecated: Do not use; this type existed for intenal-use only.  (SA1019)
model/prototest/model_test.pb.go:104:2: proto.RegisterEnum is deprecated: Use protoregistry.GlobalTypes.RegisterEnum instead.  (SA1019)
model/prototest/model_test.pb.go:105:2: proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
model/prototest/model_test.pb.go:108:15: proto.RegisterFile is deprecated: Use protoregistry.GlobalFiles.RegisterFile instead.  (SA1019)
plugin/storage/grpc/proto/storageprototest/storage_test.pb.go:8:2: Package github.com/golang/protobuf/proto is deprecated: Use the "google.golang.org/protobuf/proto" package instead.  (SA1019)
plugin/storage/grpc/proto/storageprototest/storage_test.pb.go:38:9: xxx_messageInfo_GetTraceRequest.Unmarshal is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
plugin/storage/grpc/proto/storageprototest/storage_test.pb.go:41:9: xxx_messageInfo_GetTraceRequest.Marshal is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
plugin/storage/grpc/proto/storageprototest/storage_test.pb.go:44:2: xxx_messageInfo_GetTraceRequest.Merge is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
plugin/storage/grpc/proto/storageprototest/storage_test.pb.go:47:9: xxx_messageInfo_GetTraceRequest.Size is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
plugin/storage/grpc/proto/storageprototest/storage_test.pb.go:50:2: xxx_messageInfo_GetTraceRequest.DiscardUnknown is deprecated: Do not use; this method existed for intenal-use only.  (SA1019)
plugin/storage/grpc/proto/storageprototest/storage_test.pb.go:53:37: proto.InternalMessageInfo is deprecated: Do not use; this type existed for intenal-use only.  (SA1019)
plugin/storage/grpc/proto/storageprototest/storage_test.pb.go:63:2: proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
plugin/storage/grpc/proto/storageprototest/storage_test.pb.go:66:15: proto.RegisterFile is deprecated: Use protoregistry.GlobalFiles.RegisterFile instead.  (SA1019)
proto-gen/api_v2/collector.pb.go:12:2: Package github.com/golang/protobuf/proto is deprecated: Use the "google.golang.org/protobuf/proto" package instead.  (SA1019)
proto-gen/api_v2/collector.pb.go:120:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/collector.pb.go:122:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/collector.pb.go:126:15: golang_proto.RegisterFile is deprecated: Use protoregistry.GlobalFiles.RegisterFile instead.  (SA1019)
proto-gen/api_v2/query.pb.go:14:2: Package github.com/golang/protobuf/proto is deprecated: Use the "google.golang.org/protobuf/proto" package instead.  (SA1019)
proto-gen/api_v2/query.pb.go:708:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/query.pb.go:710:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/query.pb.go:712:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/query.pb.go:714:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/query.pb.go:716:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/query.pb.go:718:2: golang_proto.RegisterMapType is deprecated: Do not use.  (SA1019)
proto-gen/api_v2/query.pb.go:720:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/query.pb.go:722:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/query.pb.go:724:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/query.pb.go:726:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/query.pb.go:728:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/query.pb.go:730:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/query.pb.go:732:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/query.pb.go:734:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/query.pb.go:738:15: golang_proto.RegisterFile is deprecated: Use protoregistry.GlobalFiles.RegisterFile instead.  (SA1019)
proto-gen/api_v2/sampling.pb.go:13:2: Package github.com/golang/protobuf/proto is deprecated: Use the "google.golang.org/protobuf/proto" package instead.  (SA1019)
proto-gen/api_v2/sampling.pb.go:397:2: golang_proto.RegisterEnum is deprecated: Use protoregistry.GlobalTypes.RegisterEnum instead.  (SA1019)
proto-gen/api_v2/sampling.pb.go:399:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/sampling.pb.go:401:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/sampling.pb.go:403:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/sampling.pb.go:405:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/sampling.pb.go:407:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/sampling.pb.go:409:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/api_v2/sampling.pb.go:413:15: golang_proto.RegisterFile is deprecated: Use protoregistry.GlobalFiles.RegisterFile instead.  (SA1019)
proto-gen/storage_v1/storage.pb.go:13:2: Package github.com/golang/protobuf/proto is deprecated: Use the "google.golang.org/protobuf/proto" package instead.  (SA1019)
proto-gen/storage_v1/storage.pb.go:801:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/storage_v1/storage.pb.go:803:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/storage_v1/storage.pb.go:805:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/storage_v1/storage.pb.go:807:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/storage_v1/storage.pb.go:809:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/storage_v1/storage.pb.go:811:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/storage_v1/storage.pb.go:813:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/storage_v1/storage.pb.go:815:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/storage_v1/storage.pb.go:817:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/storage_v1/storage.pb.go:819:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/storage_v1/storage.pb.go:821:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/storage_v1/storage.pb.go:823:2: golang_proto.RegisterMapType is deprecated: Do not use.  (SA1019)
proto-gen/storage_v1/storage.pb.go:825:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/storage_v1/storage.pb.go:827:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/storage_v1/storage.pb.go:829:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/storage_v1/storage.pb.go:831:2: golang_proto.RegisterType is deprecated: Use protoregistry.GlobalTypes.RegisterMessage instead.  (SA1019)
proto-gen/storage_v1/storage.pb.go:835:15: golang_proto.RegisterFile is deprecated: Use protoregistry.GlobalFiles.RegisterFile instead.  (SA1019)
```

Reply:
The first two warnings look relevant and should be fixed, but the rest are all from the generated code that should be excluded from linting, like we already exclude `-e model/model.pb.go`.

However, did you regenerate the files with the new version?